File: Project/Backend/src/ArtAuction.API/Controllers/ArtistController.cs
using ArtAuction.Application.Interfaces.Repositories;
using ArtAuction.Application.Interfaces.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace ArtAuction.API.Controllers;

[ApiController]
[Route("[controller]")]
[Authorize(Roles = "Admin")]
public class ArtistController : ControllerBase
{
    // Attributes
    private readonly IArtistServices _artistServices;

    // Constructor
    public ArtistController(IArtistServices artistServices)
    {
        _artistServices = artistServices;
    }
    
    // Unapproved artist API
    [HttpGet("GetUnapprovedArtists")]
    public async Task<IActionResult> GetUnapprovedArtists()
    {
        return Ok(await _artistServices.GetUnapprovedArtists());
    }
    
    // Get number of artist approved API
    [HttpGet("GetApprovedArtists")]
    public async Task<IActionResult> GetApprovedArtists()
    {
        return Ok(await _artistServices.GetApprovedArtists());
    }
    
    // Approve artist API
    [HttpPatch("ApproveArtist")]
    public async Task<IActionResult> ApproveArtist([FromQuery] int artistId, [FromQuery] int adminId)
    {
        // Check model state
        if (!ModelState.IsValid)
            return BadRequest(ModelState);
        
        // Approve artist and check result
        var result = await _artistServices.ApproveArtist(artistId, adminId);
        if (!result)
            return NotFound($"Artist {artistId} or Admin {adminId} not found or already approved");
        
        // Return success stats
        return Ok("Artist successfully approved");
    }
    
    // Reject artist
    [HttpPatch("RejectArtist")]
    public async Task<IActionResult> RejectArtist([FromQuery] int artistId)
    {
        // Check model state
        if (!ModelState.IsValid)
            return BadRequest(ModelState);
        
        // Approve artist and check result
        var result = await _artistServices.RejectArtist(artistId);
        if (!result)
            return NotFound($"Artist {artistId} not found or already approved");
        
        // Return success stats
        return Ok("Artist successfully rejected");
    }
}

File: ArtistServices/src/main/java/com/artauction/artistservices/Controllers/ArtistController.java
package com.artauction.artistservices.Controllers;

import com.artauction.artistservices.Aops.Roles.RequiresRole;
import com.artauction.artistservices.Dtos.ArtistDto;
import com.artauction.artistservices.Services.Interfaces.ArtistServices;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/Artist")
@RequiredArgsConstructor
@RequiresRole("Admin")
public class ArtistController {
    // Attributes
    private final ArtistServices artistServices;

    // Get unapproved artists API
    @GetMapping("/GetUnapprovedArtists")
    public ResponseEntity<List<ArtistDto>> getUnapprovedArtists() {
        return ResponseEntity.ok(artistServices.getUnapprovedArtists());
    }

    // Get approved artists API
    @GetMapping("/GetApprovedArtists")
    public ResponseEntity<List<ArtistDto>> getApprovedArtists() {
        return ResponseEntity.ok(artistServices.getApprovedArtists());
    }

    // Approve artist API
    @PatchMapping("/ApproveArtist")
    public ResponseEntity<?> approveArtist(@RequestParam int artistId, @RequestParam int adminId) {
        // Try to approve and check result
        boolean result = artistServices.approveArtist(artistId, adminId);
        if (!result) {
            return ResponseEntity.status(404).body("Artist " + artistId + " not found or already approved");
        }

        // Return success message
        return ResponseEntity.ok("Artist successfully approved");
    }

    // Reject artist API
    @PatchMapping("/RejectArtist")
    public ResponseEntity<?> rejectArtist(@RequestParam int artistId) {
        // Try to reject and check result
        boolean result = artistServices.rejectArtist(artistId);
        if (!result) {
            return ResponseEntity.status(404).body("Artist " + artistId + " not found");
        }

        // Return success message
        return ResponseEntity.ok("Artist successfully rejected");
    }
}